pages/rfc/sso-rfc.rst

```diff
@@ -0,0 +1,230 @@
+
+The application the user jumps to does not get notified about logout operations
+from the original application. Sessions should end after closing the browser window/tab.
+
+Terminology
+***********
+
+The following terminology is based on the OAuth 2 specification.
+
+Client Application
+==================
+The application the users starts the jump from.
+
+Resource server
+===============
+
+The application (a protected resource) the user is trying to jump to.
+
+JWT Bearer Token
+================
+
+JWT encoded Bearer token contains the user's identity, subject and custodian and
+is signed by the acting party. This token is used to obtain an OAuth 2 access token.
+
+Access Token
+============
+
+An OAuth 2 Access Token, provided by an Authorization Server. This token is handed
+to the client so it can authorize itself to a resource server. The contents of
+the token is opaque to the client. This means that the client does not need to
+know anything about the content or structure of the token itself.
+
+The resource server can exchange the Access Token at its own Authorization Server
+for a JSON document containing all the needed attributes.
+
+Authorization server
+====================
+
+The Authorization server checks the user's identity and credentials and creates the
+access token which should be used during the jump. The authorization server is
+trusted by the resource server. The resource server can exchange the access token
+for a JSON document with the user's identity, subject, custodian, validity and scopes.
+This mechanism called token introspection and is is described in `rfc7662 <https://tools.ietf.org/html/rfc7662>`_
+
+Mechanics
+*********
+
+We have two applications: The client application, and the resource server.
+Both run their own Nuts Node (could be the same node). The client application collects
+information about the user and patient, acquires an access token by posting a
+JWT Bearer Token to an authorization server (this can be the Nuts node or a
+custom implementation).
+With this access token the user jumps to the resource server. The resource server
+posts the access token to its authorization server to retrieve the contents of
+the access token.
+
+SSO Steps
+=========
+
+#. The User loads a page in the client application in the context of a certain subject (patient).
+#. The client application checks rights and settings to see if this user is allows to jump (local policy)
+#. The client application checks if the subject has any external custodians (care providers). This subject <-> custodian relationship should be registered locally.
+#. For each custodian the client application checks if it has a jump endpoint
+   The endpoint is described in the :ref:`nuts-registry-api`
+
+   .. code-block:: console
+
+     $ curl --location --request GET 'localhost:1323/api/endpoints?orgIds=urn:oid:2.16.840.1.113883.2.4.6.1:00000001&type=urn:ietf:rfc:3986:urn:oid:1.3.6.1.4.1.54851.1:nuts-sso-jump-endpoint'
```

Comment:
Double urn? Is that needed by the spec?

Reply:
I'm not sure. The `urn:ietf:rfc:3986` is the system and the `oid:1.3.6.1.4.1.54851.1:nuts-sso-jump-endpoint` the value. So, probably without the second `urn` like this: `urn:ietf:rfc:3986:oid:1.3.6.1.4.1.54851.1:nuts-sso-jump-endpoint`

Reply:
I meant type is defined by our spec as it must be a URI, the identifier is already a URI:
```
urn:oid:1.3.6.1.4.1.54851.1:nuts-sso-jump-endpoint
```
right now it is something like: URI is of type URI is of type URI is of type URI....